grouper/fe/forms.py

```diff
@@ -332,7 +332,7 @@ class UserPasswordForm(Form):
 
 class ServiceAccountCreateForm(Form):
     name = StringField("Name", [
-        validators.Length(min=3, max=32),
+        validators.Length(min=3, max=constants.MAX_NAME_LENGTH),
```

Comment:
The handler adds the @hostname before running the validation on the form, so it gets counted against the length